exotic/exotic.py

```diff
@@ -1990,16 +2014,13 @@ def main():
                         tform = transformation(np.array([imageData, firstImage]), fileName)
 
                     tx, ty = tform([exotic_UIprevTPX, exotic_UIprevTPY])[0]
-                    psf_data['target'][i] = fit_centroid(imageData, [tx, ty])
+                    psf_data['target'][i] = fit_centroid(imageData, [tx, ty], 0)
 
                     for j, coord in enumerate(compStarList):
                         ckey = f"comp{j + 1}"
 
                         cx, cy = tform(coord)[0]
-                        if cx > 0 and cy > 0:
-                            psf_data[ckey][i] = fit_centroid(imageData, [cx, cy])
-                        else:
-                            psf_data[ckey][i] = np.empty(7) * np.nan
+                        psf_data[ckey][i] = fit_centroid(imageData, [cx, cy], j+1)
```

Comment:
Does `fit_centroid(...)` return an `np.nan` by default when either of the 2-D array vals are negative? Is this something that can be safely written out of the code?

Reply:
well, the negative pixel coordinate check was ultimately a half solution, since you also get exceptions when the cx or cy exceed the positive dimensions, just the same (I've experienced these frequently).   The new logic handles both in the fit_centroid versus presuming that the caller can/will know all the conditions to be avoided.

Reply:
:heavy_check_mark: